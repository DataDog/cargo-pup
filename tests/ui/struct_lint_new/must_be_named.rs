//@compile-flags: --crate-name test_must_be_named
//@compile-flags: --crate-type lib

// This test verifies that the StructRule::MustBeNamed rule works correctly by
// flagging structs that don't match the naming pattern "Good*".

// These structs don't match the required pattern, should trigger errors
pub struct IncorrectName { //~ ERROR: Struct must match pattern 'Good*', found 'IncorrectName'
    field: i32,
}

pub struct AnotherBadName { //~ ERROR: Struct must match pattern 'Good*', found 'AnotherBadName'
    value: String,
}

// These structs match the pattern, shouldn't trigger errors
pub struct GoodName {
    data: Vec<u8>,
}

pub struct GoodStruct {
    id: u64,
    name: String,
} 