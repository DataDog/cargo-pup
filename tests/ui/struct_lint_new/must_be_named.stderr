error: Struct must match pattern 'Good*', found 'IncorrectName'
  --> tests/ui/struct_lint_new/must_be_named.rs:8:1
   |
LL | pub struct IncorrectName {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Rename this struct to match the pattern 'Good*'
   = note: Applied by cargo-pup rule 'must_be_named_test'.
   = note: `#[deny(struct_lint)]` on by default

error: Struct must match pattern 'Good*', found 'AnotherBadName'
  --> tests/ui/struct_lint_new/must_be_named.rs:12:1
   |
LL | pub struct AnotherBadName {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Rename this struct to match the pattern 'Good*'
   = note: Applied by cargo-pup rule 'must_be_named_test'.

error: aborting due to 2 previous errors

